### PR TITLE
Add assistant API and display its messages

### PR DESCRIPTION
**Pull request by sejas**
- Related https://github.com/Automattic/dotcom-forge/issues/7324
- Related https://github.com/Automattic/dotcom-forge/issues/7527

## Proposed Changes

- Fetch assistant API and render its messages

## Testing Instructions

- Run `STUDIO_AI=true npm start`
- Go to the Assistant's tab
- Send a message
- Observe the response

https://github.com/Automattic/studio/assets/779993/e6b89371-f04a-4472-8833-8dfb2cd82395

## Pre-merge Checklist

- [ ] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

**Review by jeroenpf — APPROVED** [review]
Works as described.Nice work!

I did notice that the response is not yet formatted but i assume that this is intentional so LGTM.

CleanShot 2024-05-30 at 18 46 16@2x

File: src/components/content-tab-assistant.tsx [hunk 1]
```diff
@@ -29,22 +33,36 @@ export const Message = ( { children, isUser }: MessageProps ) => (
...
+					addMessage( message, 'assistant' );
+				}
+			} catch ( error ) {
+				// A delay is added to avoid the message box being closed by the previous keydown event
+				setTimeout(
+					() =>
+						getIpcApi().showMessageBox( {
+							type: 'warning',
+							message: __( 'Failed to send message' ),
+							detail: __( "We couldn't send the latest message. Please try again" ),
```
Comment (dcalhoun) [inline src/components/content-tab-assistant.tsx]: It feels odd to punctuate one of two sentences. I presume the period absence was merely oversight. WDYT about adding a period to the second sentence?
```suggestion
							detail: __( "We couldn't send the latest message. Please try again." ),
```

**Review by dcalhoun — APPROVED** [review]
🚀

**Comment by derekblank** [comment]
> I did notice that the response is not yet formatted but i assume that this is intentional so LGTM.

Correct -- Just to connect the dots, the formatting will be addressed as part of https://github.com/Automattic/dotcom-forge/issues/7456.

File: src/components/assistant-thinking.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+				style={ { animationDelay: '0.2s' } }
+			/>
+			<div className="animate-pulse h-1.5 w-1.5 bg-a8c-blueberry rounded-full" />
+			<div
+				className="animate-pulse h-1.5 w-1.5 bg-a8c-blueberry rounded-full"
+				style={ { animationDelay: '0.4s' } }
+			/>
+		</div>
+	);
+}
```
Comment (derekblank) [inline src/components/assistant-thinking.tsx:15]: At some point I think we may want to consider grouping Assistant-related components into their own directory (or rename this component to something more generic). I think this is fine to merge as-is until we're ready for that change, however.

**Review by derekblank — APPROVED** [review]
LGTM! The testing plan succeeded for me as well, thanks for updating. Going to go ahead and merge to utilize the API calls when building other UI elements. 🚀

**Pull request merged**